tests/app/celery/test_broadcast_message_tasks.py

```diff
@@ -361,3 +409,90 @@ def test_trigger_link_tests_invokes_cbc_proxy_client(
+    broadcast_message = create_broadcast_message(sample_template)
+    current_event = create_broadcast_event(
+        broadcast_message,
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 1, 12, 1),
+    )
+    provider_message = create_broadcast_provider_message(current_event, 'ee')
+
+    check_provider_message_should_retry(provider_message)
+
+
+@freeze_time('2021-01-01 12:00')
+def test_check_provider_message_should_retry_raises_if_event_has_expired(sample_template):
+    broadcast_message = create_broadcast_message(sample_template)
+    current_event = create_broadcast_event(
+        broadcast_message,
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 1, 11, 59),
+    )
+    provider_message = create_broadcast_provider_message(current_event, 'ee')
+
+    with pytest.raises(MaxRetriesExceededError) as exc:
+        check_provider_message_should_retry(provider_message)
+    assert 'The expiry time of 2021-01-01 11:59:00 has already passed' in str(exc.value)
+
+
+@freeze_time('2021-01-01 12:00')
+def test_check_provider_message_should_retry_raises_if_a_newer_event_exists(sample_template):
+    broadcast_message = create_broadcast_message(sample_template)
+    # event approved at midnight
+    past_event = create_broadcast_event(
+        broadcast_message,
+        message_type='alert',
+        sent_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 2, 0, 0),
+    )
+    # event updated at 5am (this is the event we're currently trying to send)
+    current_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 5, 0),
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 2, 0, 0),
+    )
+    # event updated at 7am
+    future_event = create_broadcast_event(
+        broadcast_message,
+        message_type='update',
+        sent_at=datetime(2021, 1, 1, 7, 0),
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 2, 0, 0),
+    )
+    # event cancelled at 10am
+    futurest_event = create_broadcast_event(
+        broadcast_message,
+        message_type='cancel',
+        sent_at=datetime(2021, 1, 1, 10, 0),
+        transmitted_starts_at=datetime(2021, 1, 1, 0, 0),
+        transmitted_finishes_at=datetime(2021, 1, 2, 0, 0),
+    )
+
+    provider_message = create_broadcast_provider_message(current_event, 'ee')
+
+    # even though the task is going on until midnight tomorrow, we shouldn't send the update now, because the cancel
+    # message will be in the pipeline somewhere.
+    with pytest.raises(MaxRetriesExceededError) as exc:
+        check_provider_message_should_retry(provider_message)
```

Comment:
This test is only really testing that if you have a cancel event which has already started, then you should not retry the original `update`. My comments above about retry logic to do with cancelling/update might change the code but if it weren't to, then I would also like to see this test be duplicated so we see that the same behaviour happens for an update event which has already started. It's a bit confusing having both `future_event` and `futurest_event` in this test, I don't see too much benefit from having both of those fixtures here in this test